capacity4more/modules/c4m/features_og/c4m_features_og_events/c4m_features_og_events.module

```diff
@@ -219,38 +219,110 @@ function c4m_features_og_events_entity_info_alter(&$entity_info) {
  * 1 Event for anonymous user, 3 for logged in.
  *
  * @return string
- *  HTML of rendered upcoming events.
+ *  HTML of rendered upcoming events, FALSE if there's no events.
  */
 function c4m_features_og_events_upcoming_events() {
```

Comment:
This code is way to complex: 1 function is:

* Defining if we need 1 or 3 items
* If the user is logged in or not
* Query for the items
* Checking what items to load based on the filter
* Render the output
* ...

This is a recipe for bugs (way to much paths in 1 function).

I would suggest:
2 functions (1 event, 3 events for logged in users).

Each function should call other functions:

* Call a function to get the nodes.
* Call a theme function that renders the nodes in an overview with a link to the events overview.

The function that gets the events for logged in users should:
* call a helper function to get the active filter.
* call the function to collect the nodes to display with the filter param.